recipes/libpq/all/conanfile.py

```diff
@@ -134,63 +140,45 @@ def build(self):
-            config_default_pl = os.path.join(self._source_subfolder, "src", "tools", "msvc", "config_default.pl")
-            solution_pm = os.path.join(self._source_subfolder, "src", "tools", "msvc", "Solution.pm")
+            replace_in_file(self,msbuild_project_pm, "'MultiThreadedDebugDLL'", "'%s'" % runtime)
+            replace_in_file(self,msbuild_project_pm, "'MultiThreadedDLL'", "'%s'" % runtime)
+            config_default_pl = os.path.join(self.source_folder, "src", "tools", "msvc", "config_default.pl")
+            solution_pm = os.path.join(self.source_folder, "src", "tools", "msvc", "Solution.pm")
             if self.options.with_openssl:
                 for ssl in ["VC\libssl32", "VC\libssl64", "libssl"]:
-                    tools.replace_in_file(solution_pm,
+                    replace_in_file(self,solution_pm,
                                           "%s.lib" % ssl,
                                           "%s.lib" % self.deps_cpp_info["openssl"].libs[0])
                 for crypto in ["VC\libcrypto32", "VC\libcrypto64", "libcrypto"]:
-                    tools.replace_in_file(solution_pm,
+                    replace_in_file(self,solution_pm,
                                           "%s.lib" % crypto,
                                           "%s.lib" % self.deps_cpp_info["openssl"].libs[1])
-                tools.replace_in_file(config_default_pl,
+                replace_in_file(self,config_default_pl,
                                       "openssl   => undef",
                                       "openssl   => '%s'" % self.deps_cpp_info["openssl"].rootpath.replace("\\", "/"))
-            with tools.vcvars(self.settings):
-                config = "DEBUG" if self.settings.build_type == "Debug" else "RELEASE"
-                with tools.environment_append({"CONFIG": config}):
-                    with tools.chdir(os.path.join(self._source_subfolder, "src", "tools", "msvc")):
-                        self.run("perl build.pl libpq")
-                        if not self.options.shared:
-                            self.run("perl build.pl libpgport")
+
+    def build(self):
+        self._patch_sources()
+        if is_msvc(self):
+            config = "DEBUG" if self.settings.build_type == "Debug" else "RELEASE"
+            build_env = VirtualRunEnv(self)
+            env = build_env.environment()
+            env.define("CONFIG", config)
+            build_env.generate()
+
+            workdir = os.path.join(self.source_folder, "src", "tools", "msvc")
+            self.run("perl build.pl libpq", cwd=workdir)
+            if not self.options.shared:
+                self.run("perl build.pl libpgport", cwd=workdir)
         else:
-            # relocatable shared lib on macOS
-            tools.replace_in_file(
-                os.path.join(self._source_subfolder, "src", "Makefile.shlib"),
-                "-install_name '$(libdir)/",
-                "-install_name '@rpath/",
-            )
```

Comment:
if removed, it must be replaced by a call to `fix_apple_shared_install_name(self)` at the end of `package()`